src/main/java/org/aksw/iguana/cc/query/handler/QueryHandler.java

```diff
@@ -224,4 +284,67 @@ public String[] getAllQueryIds() {
     public Config getConfig() {
         return config;
     }
+
+
+   /**
+    * Instantiates pattern queries from the given query source by querying a SPARQL endpoint.
+    * A query pattern is a SPARQL 1.1 Query, which can have additional variables in the regex form of
+    * <code>%%var[0-9]+%%</code> in the Basic Graph Pattern.
+    * <p>
+    * Exemplary pattern: </br>
+    * <code>SELECT * WHERE {?s %%var1%% ?o . ?o &lt;http://exa.com&gt; %%var2%%}</code><br/>
+    * This pattern will then be converted to: <br/>
+    * <code>SELECT ?var1 ?var2 {?s ?var1 ?o . ?o &lt;http://exa.com&gt; ?var2}</code><br/>
+    * and will request query solutions from the given sparql endpoint (e.g DBpedia).<br/>
+    * The solutions will then be instantiated into the query pattern.
+    * The result may look like the following:<br/>
+    * <code>SELECT * WHERE {?s &lt;http://prop/1&gt; ?o . ?o &lt;http://exa.com&gt; "123"}</code><br/>
+    * <code>SELECT * WHERE {?s &lt;http://prop/1&gt; ?o . ?o &lt;http://exa.com&gt; "12"}</code><br/>
+    * <code>SELECT * WHERE {?s &lt;http://prop/2&gt; ?o . ?o &lt;http://exa.com&gt; "1234"}</code><br/>
+    */
+    private static List<String> instantiatePatternQueries(QuerySource querySource, Config.Pattern config) throws IOException {
+        final var patternQueries = new FileCachingQueryList(querySource);
+        final Pattern pattern = Pattern.compile("%%var\\d+%%");
```

Comment:
use `%%var[0-9]+%%` because it is easier to read and devs don't get confused when they find here a different regex than in the documentation.